.github/workflows/pr-test-lint.yml

```diff
@@ -84,6 +84,49 @@ jobs:
         name: Event File
         path: ${{ github.event_path }}
 
+  go-bench:
```

Comment:
Why do we implement this twice?

Reply:
in this file we run the benchmark for a PR. It compares the results with the results of the latest commit on `main`. The implementation for the `main` branch is different because it runs the benchmark and then uploads the benchmark results in the cache, such that PRs start comparing results with that commit.

It's a slightly different flow but I couldn't combine the 2 into a single job